### ISSUE TITLE
Leaking Rate (a)

### ISSUE THREAD

**Issue by SpyroRock** [cknd/pyESN#8]
Is there anywhere in the code the leaking parameter (a) in order to calculate the states of the network?

Thank you.

**Comment by cknd** [comment]
Hi, I'm not sure I parsed your question correctly, but there's currently no leakage parameter. In other words, the only way to control how 'excitable' the network is (i.e. how strongly and for how long it will reverberate in response to input) is by adjusting the magnitude of the recurrent weights and of the input (i.e. playing with `spectral_radius` and `input_scaling`). For example, with a lower spectral radius, activity will die out sooner.

**Comment by SpyroRock** [comment]
Thank you for your response, 
I think you answered my question!

**Comment by SpyroRock** [comment]
So, if I will try to add the leakage parameter, it would be pissible or the structure of the code doesn't allow such modification?  
The spectral radius concerns the initialization of recurrent weight. 
The leaking rate concerns how quickly the network forgets the previous states.
Right?

**Comment by cknd** [comment]
yes, that should be possible! be sure to use a default value for the new parameter that keeps the current behavior (= no decay)
